File: packages/engine/src/ai-prompts.ts

export const CANONICAL_JSON_PROMPT = `
MODE: Canonical Knowledge Generation

Given the following project idea, extract and normalize all information
into a single structured JSON object.

Input Idea:
{idea}

Instructions:
- Expand the idea logically but conservatively
- Do not invent advanced features unless implied
- Identify users, goals, scope, risks, and constraints
- If information is unclear, state assumptions explicitly
- Output ONLY valid JSON
- Do not include explanations or markdown

JSON Schema (must match exactly):
{{
  "project_overview": {{
    "summary": "",
    "problem_statement": "",
    "objectives": []
  }},
  "users": {{
    "target_users": [],
    "user_roles": []
  }},
  "scope": {{
    "in_scope": [],
    "out_of_scope": []
  }},
  "features": {{
    "must_have": [],
    "nice_to_have": []
  }},
  "constraints": {{
    "time": "",
    "budget": "",
    "technical": "",
    "regulatory": ""
  }},
  "assumptions": [],
  "risks": [],
  "success_metrics": [],
  "scalability_expectations": ""
}}
`;

export const REQUIREMENT_PROMPT = `
MODE: Document Generation

Using ONLY the provided canonical JSON,
generate a professional Requirement Specification document.

Rules:
- Do not introduce new features
- Use formal documentation tone
- Structure the document clearly
- All requirements must map to in_scope and must_have features

Canonical JSON:
{canonical_json}

Document Structure:
- Purpose
- Scope
- User Roles
- Functional Requirements
- Non-Functional Requirements
- Assumptions
- Constraints
- Dependencies
`;

export const PLANNING_PROMPT = `
MODE: Document Generation

Using ONLY the provided canonical JSON,
generate a project Planning document.

Rules:
- Base milestones on scope and features
- Reflect stated constraints
- Avoid technical implementation details

Canonical JSON:
{canonical_json}

Document Structure:
- Project Goals
- Deliverables
- Project Phases
- Milestones
- Resource Overview
- Communication Plan
`;

export const ARCHITECTURE_PROMPT = `
MODE: Document Generation

Using ONLY the provided canonical JSON,
generate a high-level Technical Architecture document.

Rules:
- Choose standard, proven technologies
- Avoid unnecessary complexity
- Justify technical choices briefly
- Align with scalability expectations

Canonical JSON:
{canonical_json}

Document Structure:
- System Overview
- High-Level Architecture
- Technology Stack
- Data Flow Overview
- Security Considerations
- Scalability Strategy
- Deployment Overview
`;

export const IPMP_PROMPT = `
MODE: Document Generation

Using ONLY the provided canonical JSON,
generate an Integrated Project Management Plan (IPMP).

Rules:
- Reference planning and governance
- Assume standard enterprise controls
- Avoid repetition of full requirements

Canonical JSON:
{canonical_json}

Document Structure:
- Project Overview
- Governance Structure
- Scope Management
- Schedule Management
- Cost Management
- Quality Management
- Risk Management
- Change Control
- Communication Management
`;

export const SCHEDULE_COST_PROMPT = `
MODE: Document Generation

Using ONLY the provided canonical JSON,
generate a Schedule and Cost Estimation document.

Rules:
- Use conservative estimates
- Clearly separate time and cost
- Include contingency buffer

Canonical JSON:
{canonical_json}

Document Structure:
- Work Breakdown Structure (High-Level)
- Estimated Timeline
- Resource Effort Estimates
- Cost Breakdown
- Contingency Planning
`;

export const QUALITY_RISK_PROMPT = `
MODE: Document Generation

Using ONLY the provided canonical JSON,
generate a Quality Assurance and Risk Management document.

Rules:
- Risks must come from canonical risks or logical implications
- Include mitigation strategies
- Define quality acceptance clearly

Canonical JSON:
{canonical_json}

Document Structure:
- Quality Objectives
- Quality Standards
- Review & Approval Process
- Risk Register
- Risk Mitigation Plan
`;

export const TESTING_RELEASE_PROMPT = `
MODE: Document Generation

Using ONLY the provided canonical JSON,
generate a Testing Strategy document.

Rules:
- Testing must align with requirements and risks
- No implementation-specific test cases
- Focus on validation and acceptance

Canonical JSON:
{canonical_json}

Document Structure:
- Test Strategy
- Test Types
- Test Scenarios (High-Level)
- Entry & Exit Criteria
- Defect Management
- Release Approval
`;

export const UI_UX_PROMPT = `
MODE: Document Generation

Using ONLY the provided canonical JSON,
generate a Detailed UI/UX Design Specification.

Rules:
- Generate 5-7 key screens based on the features and user roles
- Use professional design terminology
- CRITICAL: YOUR OUTPUT MUST BE PARSABLE. USE THIS FORMAT FOR EACH SCREEN:

### [Screen Name] Screen
- **Purpose**: A clear description of why this screen exists.
- **Roles**: Which user roles can access this.
- **Components**: A comma-separated list of UI components (e.g., Header, Sidebar, Search Bar, Card List).
- **Interactions**: User actions available (e.g., Click 'Buy' to add to cart, Drag and Drop to reorder).
- **States**: Different UI states (e.g., Loading, Empty, Error, Success).

Canonical JSON:
{canonical_json}
`;

File: apps/api-py/engine/ai_prompts.py
CANONICAL_JSON_PROMPT = """
You are a Product Strategy Expert. 
Your task is to transform a high-level project idea into a **Structured Canonical JSON Object**.
This JSON will serve as the single source of truth for all downstream documentation.

**Project Idea**: {idea}

**CRITICAL RULES:**
1.  **Output MUST be ONLY valid JSON**. No markdown formatting, no preamble.
2.  **Schema Consistency**: You must adhere to the structure below.
3.  **Expansion**: Expand the idea realistically. If an idea is "A gym app", define user roles (Trainer, Member), core features (Workout logging, etc.), and risks.
4.  **Conservative Scope**: Do not over-engineer. Focus on core MVP requirements.

**REQUIRED JSON SCHEMA:**
{{
  "project_overview": {{
    "summary": "Detailed 2-3 sentence overview",
    "problem_statement": "The specific problem this solves",
    "objectives": ["Goal 1", "Goal 2", "Goal 3"]
  }},
  "users": {{
    "target_users": ["User Type A", "User Type B"],
    "user_roles": ["Admin", "Standard User", "etc."]
  }},
  "scope": {{
    "in_scope": ["Feature A", "Feature B", "Module C"],
    "out_of_scope": ["Feature X", "Future expansion Y"]
  }},
  "features": {{
    "must_have": ["Core Feature 1", "Core Feature 2"],
    "nice_to_have": ["Extra 1", "Extra 2"]
  }},
  "constraints": {{
    "time": "e.g., 12 weeks",
    "budget": "e.g., Enterprise standard",
    "technical": "e.g., Cloud-native",
    "regulatory": "e.g., GDPR"
  }},
  "assumptions": ["Assumption 1", "Assumption 2"],
  "risks": ["Risk 1", "Risk 2"],
  "success_metrics": ["Metric 1", "Metric 2"],
  "scalability_expectations": "e.g., Support 10k concurrent users"
}}
"""

REQUIREMENT_PROMPT = """
You are a PMP-certified Software Project Consultant.
Create a **Requirement Document** using the Canonical JSON provided.

**Canonical JSON**: {canonical_json}

**CRITICAL: Use these 14 sections.**
1. Document Control & Versioning
2. Project Background & Objectives
3. Stakeholder Identification
4. Assumptions & Constraints
5. In-Scope / Out-of-Scope
6. Functional Requirements (REQ-001 format)
7. Non-Functional Requirements (NFR-001 format)
8. User Roles & Permissions
9. Business Rules
10. Use Cases / User Stories
11. UI/UX & Screen References
12. Data Requirements
13. Regulatory & Compliance Requirements
14. Requirement Traceability Matrix (RTM)
"""

PLANNING_PROMPT = """
You are a PMP-certified Software Project Consultant.
Create a **Project Planning Document** using the Canonical JSON provided.

**Canonical JSON**: {canonical_json}

**CRITICAL: Use these 11 sections.**
1. Project Overview
2. Project Governance Structure
3. Project Organization & Roles
4. Project Methodology (Agile Scrum recommended)
5. Work Breakdown Structure (WBS)
6. Deliverables & Milestones
7. Resource Planning
8. Communication Management Plan
9. Change Management Plan
10. Dependency Management
11. Assumptions & Constraints
"""

ARCHITECTURE_PROMPT = """
You are a Senior System Architect.
Create a **Technical Architecture & Delivery Plan** using the Canonical JSON provided.

**Canonical JSON**: {canonical_json}

**CRITICAL: Use these 12 sections.**
1. Architecture Overview
2. System Context Diagram
3. Logical Architecture
4. Physical Architecture
5. Technology Stack (Next.js, Node.js, Python, PostgreSQL)
6. Application Architecture
7. Database & Data Flow Design
8. API & Integration Strategy
9. Security Architecture
10. Scalability & Performance Design
11. Deployment Architecture
12. Technical Risks & Mitigations
"""

IPMP_PROMPT = """
You are a PMP-certified Project Manager.
Create an **Integrated Project Management Plan (IPMP)** using the Canonical JSON provided.

**Canonical JSON**: {canonical_json}

**CRITICAL: Use these 11 sections.**
1. IPMP Purpose & Scope
2. Project Objectives & Success Criteria
3. Integrated Baselines (Scope, Schedule, Cost)
4. Governance & Decision Framework
5. Integrated Change Control
6. Risk, Quality & Procurement Integration
7. Stakeholder Engagement Strategy
8. Performance Measurement (KPIs, EV, Metrics)
9. Reporting & Review Cadence
10. Escalation & Issue Resolution
11. Compliance & Audit Strategy
"""

SCHEDULE_COST_PROMPT = """
You are a Project Scheduler and Cost Controller.
Create a **Schedule & Cost Plan** using the Canonical JSON provided.

**Canonical JSON**: {canonical_json}

**CRITICAL: Use these 11 sections.**
1. Schedule Management Approach
2. Project Timeline & Milestones
3. Task Dependencies
4. Resource Allocation
5. Critical Path Analysis
6. Cost Estimation Methodology
7. Budget Breakdown (CAPEX / OPEX)
8. Cost Baseline
9. Cost Control & Tracking
10. Earned Value Management (EVM)
11. Schedule & Cost Risks
"""

QUALITY_RISK_PROMPT = """
You are a Quality and Risk Manager.
Create a **Quality, Risk & Procurement Plan** using the Canonical JSON provided.

**Canonical JSON**: {canonical_json}

**CRITICAL: Use these 15 sections.**
1. Quality Management Approach
2. Quality Objectives
3. Quality Standards & Metrics
4. Quality Assurance Process
5. Quality Control Activities
6. Risk Management Approach
7. Risk Identification
8. Risk Register
9. Risk Analysis & Prioritization
10. Risk Response Strategies
11. Procurement Management Approach
12. Procurement Strategy
13. Vendor Selection Criteria
14. Contract Types
15. SLA & Performance Monitoring
"""

TESTING_RELEASE_PROMPT = """
You are a QA Lead and Release Manager.
Create a **Testing & Release Plan** using the Canonical JSON provided.

**Canonical JSON**: {canonical_json}

**CRITICAL: Use these 12 sections.**
1. Test Strategy
2. Test Scope & Objectives
3. Test Environment Setup
4. Test Types (Unit, Integration, System, UAT)
5. Test Data Management
6. Defect Management Process
7. Entry & Exit Criteria
8. Release Management Strategy
9. Deployment Plan
10. Rollback & Recovery Plan
11. Post-Release Validation
12. Maintenance & Support Strategy
"""

UI_UX_PROMPT = """
You are a Senior UI/UX Designer.
Create a **Detailed UI/UX Design Specification** using the Canonical JSON provided.

**Canonical JSON**: {canonical_json}

**CRITICAL: Use these 5 sections.**
1. Design System & Style Guide
2. User Flows
3. Screen Specifications (Purpose, Roles, Components, Interactions, States)
4. Accessibility Standards
5. Mobile Responsiveness
"""
